File: 11-container-with-most-water/11-container-with-most-water.js
/**
 * @param {number[]} height
 * @return {number}
 */
const maxArea = function(height) {
    let max = 0,
        left = 0,
        right = height.length - 1;
    while (left < right) {
        let current = (right - left) * Math.min(height[left], height[right]);
        max = Math.max(max, current);
        if (height[left] < height[right]) {
            left++;
        } else {
            right--;
        }
    }
    return max;
};

File: 11-container-with-most-water/11-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        max_area, l, r = 0, 0, len(height) - 1
        while l < r:
            area = (r - l) * (min(height[l], height[r]))
            max_area = max(max_area, area)
            if height[l] < height[r]:
                l += 1
            else:
                r -= 1
        return max_area